report.qmd

```diff
@@ -212,13 +467,391 @@ Using the Saaty matrix, each criteria are weighted against each other to give ea
+
+#head(grids)
+
+# Show how many grids fall into each cluster
+#print(table(grids$cluster))
+
+##########################################################
+# Run the K means Clustering
+##########################################################
+
+library(RColorBrewer)
+
+# Choose a palette and number of clusters
+k <- length(unique(grids$cluster))  
+my_colors <- brewer.pal(n = k, name = "Spectral") 
+
+# Assign colors to clusters
+cluster_colors <- my_colors
+names(cluster_colors) <- as.character(1:k)
+
+# Create color vector for plotting
+plot_colors <- cluster_colors[as.character(as.numeric(grids$cluster))]
+
+# Plot using base R
+plot(st_geometry(grids), 
+     col = plot_colors,
+     main = "Spatial Pattern of Urban Stream Clusters", 
+     border = NA)
+
+# Add legend
+legend("topright", 
+       legend = paste("Cluster", names(cluster_colors)), 
+       fill = cluster_colors, 
+       title = "Clusters",
+       bty = "n",
+       cex = 0.8)
+```
+
+To be able the visualize the distribution of the 5 clusters in 3D space, the following plot was created. As we have 4 indicators, and we cannot easily visualize the 4 dimensional representation of our clustering. When deciding on which indicators to use for the axis, we inspected all possible options and chose the most appealing visual representation which spatially distributed the clusters the best.
+
+```{r echo=FALSE, warning=FALSE, message=FALSE}
+
+##########################################################
+# Visualise the 3d plot of the Kmeans cluster
+##########################################################
+
+# For 3D plotly
+
+features$cluster <- grids$cluster
+features$color <- cluster_colors[as.character(features$cluster)]
+
+features$color <- cluster_colors[as.character(features$cluster)]
+
+plot_ly(
+  data = features,
+  x = ~Percentage_habitat_n,
+  y = ~NDVI_norm,
+  z = ~AccesstoStream_norm,
+  color = ~cluster,
+  colors = cluster_colors,
+  type = "scatter3d",
+  mode = "markers",
+  marker = list(size = 3)
+) %>%
+  layout(
+    title = "3D Cluster Plot: Green-Blue Functionality Typology",
+    scene = list(
+      xaxis = list(title = "Proportion of Natural Habitat"),
+      yaxis = list(title = "NDVI (Greenness)"),
+      zaxis = list(title = "Access to Stream")
+    )
+  )
+```
+Therefore, to ensure another representation of each cluster in respect to their indicator we can plot Polar charts. This can help visualise the 4 dimensional nature and what typological counterpart each cluster would represent. Based from these charts we can assign each cluster a correct typology.
+
+```{r echo=FALSE, warning=FALSE, message=FALSE}
+
+##########################################################
+# Create Radar Charts for the 5 Typologies
+##########################################################
+
+# Install and load necessary packages
+if (!require("fmsb")) install.packages("fmsb")
```

Comment:
I know you are installing this conditionally, but quarto might complain at render time about this, so it is better to simply add `library(fmsb)` in the same place where you have all library calls in the beginning. The user will know that the package needs to be installed because the notebook will not render otherwise.